codice/ddf
# DDF-04556 Generate Solr stop key

##### ABBREVIATED REVIEW BETWEEN 2.13.X AND MASTER IS IN EFFECT
Link to 2.13.x PR #4562 
Link to 2.13.x PR #4506 
Link to 2.13.x PR #4433

#### What does this PR do?
Generates a unique stop key for every invocation of the Solr process.

#### Who is reviewing it? 
@austinsteffes @tyler30clemens 
@codice/solr 

@brjeter
@clockard
@vinamartin 

#### How should this be tested?
Test cases are below. Additionally, feel free to do some exploratory testing like starting solr twice without a stopping in between.

Test 1 - Mac
- Precondition: Solr is NOT running
- Precondition: Terminal in the DDF bin directory
- Run `ddfsolr.bat start`
- Expected result 1: Solr is running
- Expected result 2: A file named STOPKEY is in the bin directory and it is 8 bytes long
- Run `ddfsolr.bat stop`
- Expected result 1: Solr is NOT running
- Expected result 2: The STOPKEY has been removed

Test 2 - Mac
- Precondition: Solr and DDF are NOT running
- Precondition: Terminal in the DDF bin directory
- Run `ddf.bat start`
- Expected result 1: The DDF starts and the DDF log shows a successful connection to Solr
- Expected result 2: A file named STOPKEY is in the bin directory and it is 8 bytes long
- Shutdown DDF
- Expected result 1: Solr and DDF are NOT running
- Expected result 2: The STOPKEY has been removed

Test 3 - Windows
- Precondition: Solr is NOT running
- Precondition: Terminal in the DDF bin directory
- Run `ddfsolr.bat start`
- Expected result 1: Solr is running
- Expected result 2: A file named STOPKEY is in the bin directory and it is 8 bytes long
- Run `ddfsolr.bat stop`
- Expected result 1: Solr is NOT running
- Expected result 2: The STOPKEY has been removed

Test 4 - Windows
- Precondition: Solr and DDF are NOT running
- Precondition: Terminal in the DDF bin directory
- Run `ddf.bat start`
- Expected result 1: The DDF starts and the DDF log shows a successful connection to Solr
- Expected result 2: A file named STOPKEY is in the bin directory and it is 8 bytes long
- Shutdown DDF
- Expected result 1: Solr and DDF are NOT running
- Expected result 2: The STOPKEY has been removed

For GH Issues:
Fixes: #4563 

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either ainline line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
